staging/src/kubevirt.io/api/instancetype/v1beta1/types.go

```diff
@@ -592,6 +591,11 @@ type PreferenceRequirements struct {
 	//
 	//+optional
 	Memory *MemoryPreferenceRequirement `json:"memory,omitempty"`
+
+	// Specifies the required architecture of the vm.
+	//
+	//+optional
+	Architecture *string `json:"architecture,omitempty"`
```

Comment:
It's a string in the VMI spec too, so IMO it's fine to keep it a string. But what about adding an admitting webhook to validate the provided value?